### PR TITLE
Ctcp 4855

### PR DESCRIPTION
**Pull request by malaw-hmrc**

### CODE REVIEW

File: app/utils/answersHelpers/ConsignmentAnswersHelper.scala [hunk 1]
```diff
@@ -17,21 +17,24 @@
...
 import viewModels.sections.Section.{AccordionSection, StaticSection}
+import play.api.mvc.Call
 
 class ConsignmentAnswersHelper(userAnswers: UserAnswers)(implicit messages: Messages) extends AnswersHelper(userAnswers) {
 
   def headerSection: Section = StaticSection(
     rows = Seq(
       traderAtDestinationRow
-    )
+    ) ++ grossMass
```
Comment (jackheslop96) [inline app/utils/answersHelpers/ConsignmentAnswersHelper.scala:37]:
```
rows = Seq(
    Some(traderAtDestinationRow),
    grossMass
).flatten
```

File: app/pages/grossMass/GrossMassPage.scala [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ * limitations under the License.
+ */
+
+package pages.grossMass
+
+import models.Index
+import pages.QuestionPage
+import play.api.libs.json.JsPath
+
+case object GrossMassPage extends QuestionPage[String] {
```
Comment (jackheslop96) [inline app/pages/grossMass/GrossMassPage.scala:23]: Should be `QuestionPage[BigDecimal]`

File: conf/messages.en [hunk 3]
```diff
@@ -590,4 +590,8 @@ additionalReference.index.additionalReferenceType.CheckMode.heading = What is th
 additionalReference.index.additionalReferenceType.error.required = Select the type of additional reference
 additionalReference.index.additionalReferenceType.check.error.required = Select the new additional reference type
 additionalReference.index.additionalReferenceType.hint = Enter the reference name or code, like Carrier (AEO certificate number) or Y028.
-additionalReference.index.additionalReferenceType.placeholder = Select an additional reference
\ No newline at end of file
+additionalReference.index.additionalReferenceType.placeholder = Select an additional reference
+unloadingFindings.consignment.heading = Consignment
```
Comment (jackheslop96) [inline conf/messages.en:594]: Not used?

File: app/utils/transformers/GrossMassTransformer.scala [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+class GrossMassTransformer @Inject() (implicit ec: ExecutionContext) extends PageTransformer {
+
+  def transform(grossMass: Option[BigDecimal]): UserAnswers => Future[UserAnswers] = userAnswers => {
+    grossMass match {
+      case Some(value) =>
+        lazy val pipeline: UserAnswers => Future[UserAnswers] = set(GrossMassPage, value.toString())
+        pipeline(userAnswers)
+      case None => Future.successful(userAnswers)
+    }
+  }
```
[the remark this answers is not in the document]
Comment (malaw-hmrc) [reply app/utils/transformers/GrossMassTransformer.scala]: Done

File: conf/messages.en (hunk 3 above)
Comment (malaw-hmrc) [reply conf/messages.en]: Yep not used and removed

File: app/pages/grossMass/GrossMassPage.scala (hunk 2 above)
Comment (malaw-hmrc) [reply app/pages/grossMass/GrossMassPage.scala]: Refactored

File: app/utils/transformers/ConsignmentTransformer.scala [hunk 5]
```diff
@@ -42,10 +43,20 @@ class ConsignmentTransformer @Inject() (
             departureTransportMeansTransformer.transform(consignment05.DepartureTransportMeans) andThen
             documentsTransformer.transform(consignment05.SupportingDocument, consignment05.TransportDocument) andThen
             houseConsignmentsTransformer.transform(consignment05.HouseConsignment) andThen
-            additionalReferenceTransformer.transform(consignment05.AdditionalReference)
+            additionalReferenceTransformer.transform(consignment05.AdditionalReference) andThen
+            transformGrossMass(consignment05.grossMass)
```
[the remark this answers is not in the document]
Comment (malaw-hmrc) [reply app/utils/transformers/ConsignmentTransformer.scala]: refactored

**Review by jackheslop96 — APPROVED** [review]

**Review by joecefnhedog — DISMISSED** [review]

**Pull request merged**